arangod/Aql/Query.cpp

```diff
@@ -183,8 +182,8 @@ Query::Query(std::shared_ptr<transaction::Context> ctx, QueryString queryString,
             std::move(options),
             std::make_shared<SharedQueryState>(ctx->vocbase().server())) {}
 
-/// @brief destroys a query
 Query::~Query() {
+  TRI_ASSERT(_queryProfile == nullptr);
```

Comment:
Maybe add a comment here that the _most derived_ class needs to explicitly reset `_queryProfile` in order to unregister the query from the query list, because otherwise we have potential data races.

Reply:
Good point

Reply:
Done